x/ibc/04-channel/port.go

```diff
@@ -0,0 +1,37 @@
+package channel
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	commitment "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+)
+
+type Port struct {
```

Comment:
This should use opaque memory references. See how store keys work in the SDK - the same mechanism is appropriate.